homeassistant/helpers/device_registry.py

```diff
@@ -38,6 +38,17 @@ class DeviceEntry:
     id = attr.ib(type=str, default=attr.Factory(lambda: uuid.uuid4().hex))
 
 
+def format_mac(mac):
+    """Format the mac address string for entry into dev reg."""
+    lower_mac = mac.lower()
+    if len(lower_mac) == 12:
+        # no : included
+        return ':'.join(lower_mac[i:i + 2] for i in range(0, 12, 2))
```

Comment:
Wouldn't it be better to strip everything but hexadecimal values from the Mac? There are multiple standards to format the Mac. Also since if one implementation modifies its Mac to follow its own standard the smallest common denominator should be used.

Reply:
We are normalizing the value here to lowercase and separated by `:`. If an integration sends us any other format, it will be reformatted to ours, so no need to stick to a common denominator because we specify the common one :)

Reply:
It only normalise Mac that doesn't have any ':' but doesn't cover Mac that uses '.' or '-' as separators, see https://en.wikipedia.org/wiki/MAC_address

Reply:
Okay, I'll strip out any non hexadecimal value and then format it with `:`?